overlord/snapstate/snapstate.go

```diff
@@ -632,6 +636,10 @@ func Install(st *state.State, name, channel string, revision snap.Revision, user
 		return nil, err
 	}
 
+	if flags.Classic && !info.NeedsClassic() {
+		// snap does not require classic confinement, silently drop the flag
+		flags.Classic = false
+	}
 	if err := checkInstallPreconditions(st, info, flags, &snapst); err != nil {
```

Comment:
can we add a TODO about trying to integrate with the helper later (it's probably not trivial for various reasons but slightly appealing)